lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1203,42 +1209,81 @@ def cmd_route_tabs(str, words)
     ret
   end
 
+  # Print save help information
   def cmd_save_help
-    print_line "Usage: save"
-    print_line
-    print_line "Save the active datastore contents to disk for automatic use across restarts of the console"
+    print_line 'Usage: save [options]'
     print_line
+    print_line 'Save the active datastore contents to disk for automatic use across restarts of the console'
     print_line "The configuration is stored in #{Msf::Config.config_file}"
-    print_line
+    print @@save_opts.usage
   end
 
   #
-  # Saves the active datastore contents to disk for automatic use across
+  # Deletes or saves the active datastore contents to disk for automatic use across
   # restarts of the console.
   #
   def cmd_save(*args)
-    # Save the console config
-    driver.save_config
-
-    begin
-      FeatureManager.instance.save_config
-    rescue StandardException => e
-      elog(e)
+    if args.include?('-h') || args.include?('--help')
+      cmd_save_help
+      return
     end
 
-    # Save the framework's datastore
-    begin
-      framework.save_config
+    if args.empty?
+      # Save config for current module
+      # Save the console config
+      driver.save_config
 
-      if (active_module)
-        active_module.save_config
+      begin
+        FeatureManager.instance.save_config
+      rescue StandardException => e
+        elog(e)
       end
-    rescue
-      log_error("Save failed: #{$!}")
-      return false
+
+      # Save the framework's datastore
+      begin
+        framework.save_config
+
+        if active_module
+          active_module.save_config
+        end
+      rescue
+        log_error("Save failed: #{$!}")
+        return false
+      end
+
+      print_line("Saved configuration to: #{Msf::Config.config_file}")
     end
 
-    print_line("Saved configuration to: #{Msf::Config.config_file}")
+    @@save_opts.parse(args) do |opt|
+      case opt
+      when '-d'
+        # Delete all saved options for modules from the config file.
+        # No framework options will be deleted.
+        begin
+          ini = Rex::Parser::Ini.new(::Msf::Config.config_file)
+
+          ini.delete_if { |k, _v| !k.start_with?('framework') }
+
+          ini.to_file(::Msf::Config.config_file)
+        rescue StandardError
+          print_error("Failed to delete console config: #{$!}")
+        end
+
+        if active_module
+          active_module.import_defaults
+        end
+        print_line('Deleted saved configs for all modules.')
+      when '-r'
+        active_module.import_defaults
```

Comment:
This doesn't seem to work for payload modules, unless I've misunderstood  🤔 
```
msf6 payload(python/meterpreter/reverse_tcp) > options

Module options (payload/python/meterpreter/reverse_tcp):

   Name   Current Setting  Required  Description
   ----   ---------------  --------  -----------
   LHOST  192.168.123.1    yes       The listen address (an interface may be specified)
   LPORT  4444             yes       The listen port

msf6 payload(python/meterpreter/reverse_tcp) > save -r
Reloaded default options.
msf6 payload(python/meterpreter/reverse_tcp) > options

Module options (payload/python/meterpreter/reverse_tcp):

   Name   Current Setting  Required  Description
   ----   ---------------  --------  -----------
   LHOST  192.168.123.1    yes       The listen address (an interface may be specified)
   LPORT  4444             yes       The listen port
```

It does seem to work for exploits, so I imagine this is an unrelated bug? 👀